internal/k8shandler/cluster.go

```diff
@@ -393,3 +400,36 @@ func addNodeState(node NodeTypeInterface, nodeStatus *api.ElasticsearchNodeStatu
 	nodeStatus.DeploymentName = nodeState.DeploymentName
 	nodeStatus.StatefulSetName = nodeState.StatefulSetName
 }
+
+func (er *ElasticsearchRequest) checkWatermarkAndUnblockIndices() {
+	er.refreshDiskWatermarkThresholds()
+	if er.isDiskUtilizationBelowThreshold() {
+		indices, err := er.esClient.GetAllIndices("")
+		if err != nil {
+			log.Error(err, "failed to fetch all indices")
+		}
+		for _, index := range indices {
+			if index.Index == securityIndex {
+				continue
+			}
+			isBlocked := er.isIndexBlocked(index.Index)
+			if isBlocked {
+				er.unblockIndex(index.Index)
+			}
+		}
+	}
+}
+
+func (er *ElasticsearchRequest) isDiskUtilizationBelowThreshold() bool {
```

Comment:
can we rename this to use "FloodWatermark" instead of "Threshold" to be more descriptive?